### DISCUSSION TITLE
change the color of imported kml track

### DISCUSSION THREAD

**Discussion by krzsiek** [organicmaps/organicmaps#1092] [Ideas]
Can you add options to change the color of the displayed track imported from kml? My all added tracks now display in dark blue. In the track info menu there is a circle with the track color and it would be nice if you click on it to be able to change the color of what you see on the map. 

Now with impoted several tracks of the same color displayed, everything becomes unreadable.

**Comment by biodranik** [comment]
Can you please share your KML so we can understand why the original color was not recognized (email is ok if you don't want to share publicly)? Or tracks were originally of the same color?

**Comment by krzsiek** [comment]
Thanks for the answer.

The issue does not concern the recognition of the color stored in the
file.  I am using a gpx to kml converter, and having kml files I would like
to be able to easily change the color in the application after importing
the kml file into the program.

It is easier to change what the apk displays than to go back to the kml
editor, change track parameters, export and import it back to the Organic
Map.

Best regards.

ps
Also thanks for the option to import bookmarks.  This was missing in mapsme
before.

**Reply by biodranik** [reply]
So initially, before the import, colors were also dark blue?

I understand your request, but want to make sure we don't have other bugs here.

**Comment by krzsiek** [comment]
Yes, the color interpretation is ok. Another application gives a different
color and I have no problems with that.  It is about the ability to change
the display color regardless of what is defined in the imported file.

Something like it has always been in various Garmin navigation devices.

 Best regards

**Comment by krzsiek** [comment]
Hi, here are some examples of kml file

This file has a color definition and Organicmaps displays it as pink as it
is defined.


`
<?xml version="1.0" encoding="UTF-8" standalone="yes"?>
<kml xmlns="http://earth.google.com/kml/2.1">
    <Document>
        <open>true</open>
        <Style id="routeStyle">
            <LineStyle>
                <color>7FFF0055</color>
                <width>3.0</width>
            </LineStyle>
        </Style>
        <Style id="trackStyle">
            <LineStyle>
                <color>FFFF00FF</color>
                <width>3.0</width>
            </LineStyle>
        </Style>
        <Placemark>
            <name>Track: Trasa2020 dz1</name>
            <styleUrl>#trackStyle</styleUrl>
            <LineString>
                <coordinates>20.71.......................
'

This converted file without a color definition is displayed with a default
dark blue color.


`
<kml>
   <Document>
      <name/>
      <Placemark>
         <description></description>
         <name>new</name>
         <LineString>
            <coordinates>22.0.................
`


The problem is precisely those files with no color definition, all
displayed the same color.

Such files exist and you have to live with them somehow.

Therefore, the option to change the color on the Organicmaps  would make
the use of the apk very much easier.

Regards

**Comment by epikgit** [comment]
I have the exact same problem and I support the same exact solution: a feature to change the kml default blue color.

Since I like very much cycling around my place I had a lot of overlapping tracks all in blue... and it's a mess.
![image](https://user-images.githubusercontent.com/14310321/135464500-eb146b1b-a773-46df-a2be-d60f1ceff974.png)


I know that I could activate only the track I want to follow, but I like to change my mind during the trip and move from the track 1 to the track 5, and I can't stop the bike to switch them on the phone.

Anyway thank you for this terrific app. Maps.me has become too much heavy and battery draining for my taste.

**Comment by biodranik** [comment]
Changing track color is a useful feature. Isn't it already listed somewhere in open issues?

**Comment by alanwilter** [comment]
Any updates on this? It would be really really great to be able to change the colours within the app.

**Comment by mixmaxmux** [comment]
I make the same request for GPX files.

In addition, it would be nice to distinguish the starting point from the arrival point and indicators that indicate the direction of the route.

I have gpx loop routes and I can't tell where they start from.

Thank you for this wonderful app.

**Comment by biodranik** [comment]
Changing the imported track (GPX or KML) color for iOS has been implemented and released. Any volunteers to do it in a similar way for Android are welcome! Some examples of how it works on iOS are here: https://github.com/organicmaps/organicmaps/pull/7499

**Reply by alanwilter** [reply]
I tested on iOS and it's working beautifully! Many thanks for that.
